### PR TITLE
Unify receive/ and receive_out_of_band/ endpoints

### PR DESCRIPTION
**Pull request by Jakub89**
Part 1 of: #787

### CODE REVIEW

File: concent_api/core/message_handlers.py [hunk 1]
```diff
@@ -908,18 +908,21 @@ def store_subtask(
...
-    assert client_public_key    not in ['', None]
+def handle_messages_from_database(client_public_key: bytes) -> Union[message.Message, None]:
+    assert client_public_key not in ['', None]
 
     encoded_client_public_key = b64encode(client_public_key)
-    pending_response = PendingResponse.objects.select_for_update().filter(
-        client__public_key = encoded_client_public_key,
-        queue              = response_type.name,
-        delivered          = False,
+    pending_response = PendingResponse.objects.filter(
```
Comment (pawelkisielewicz) [inline concent_api/core/message_handlers.py]: are you sure we should remove `select_for_update` in this place? What is the reason?

File: concent_api/core/message_handlers.py [hunk 2]
```diff
@@ -908,18 +908,21 @@ def store_subtask(
...
+        client__public_key=encoded_client_public_key,
+        queue=PendingResponse.Queue.Receive.name,  # pylint: disable=no-member
+        delivered=False,
     ).order_by('created_at').first()
+    if pending_response is None:
+        pending_response = PendingResponse.objects.filter(
```
Comment (pawelkisielewicz) [inline concent_api/core/message_handlers.py]: no `select_for_update` here?

File: concent_api/core/message_handlers.py (hunk 1 above)
Comment (Jakub89) [reply concent_api/core/message_handlers.py]: No reason, I did it by accident on rebase.

File: concent_api/core/message_handlers.py [hunk 3]
```diff
@@ -908,18 +908,21 @@ def store_subtask(
...
+    assert client_public_key not in ['', None]
 
     encoded_client_public_key = b64encode(client_public_key)
-    pending_response = PendingResponse.objects.select_for_update().filter(
-        client__public_key = encoded_client_public_key,
-        queue              = response_type.name,
-        delivered          = False,
+    pending_response = PendingResponse.objects.filter(
+        client__public_key=encoded_client_public_key,
+        queue=PendingResponse.Queue.Receive.name,  # pylint: disable=no-member
```
Comment (rwrzesien) [inline concent_api/core/message_handlers.py]: Do we still need `queue` field and `PendingResponse.Queue` enum ?

File: concent_api/core/tests/test_auth_report_computed_task.py [hunk 4]
```diff
@@ -920,16 +920,6 @@ def test_provider_forces_computed_task_report_and_requestor_sends_rejection_due_
         self.assertEqual(response.status_code,        204)
         self.assertEqual(len(response.content),       0)
 
-        with freeze_time("2017-12-01 11:00:15"):
```
Comment (rwrzesien) [inline concent_api/core/tests/test_auth_report_computed_task.py]: Why is this removed ?

File: concent_api/core/message_handlers.py (hunk 3 above)
Comment (Jakub89) [reply concent_api/core/message_handlers.py]: I'm pretty sure there still need to be distinction between "standard" messages and "out-of-band".

File: concent_api/core/tests/test_auth_report_computed_task.py (hunk 4 above)
Comment (Jakub89) [reply concent_api/core/tests/test_auth_report_computed_task.py]: Because this part return "out-of-band" message that we check later, so I don't think we need to replace it with different call.

File: concent_api/core/message_handlers.py (hunk 3 above)
Comment (rwrzesien) [reply concent_api/core/message_handlers.py]: As discussed, lets keep it.

File: concent_api/core/tests/test_auth_report_computed_task.py (hunk 4 above)
Comment (rwrzesien) [reply concent_api/core/tests/test_auth_report_computed_task.py]: In other words, I think it should stay it if doesn't break anything and test one more corner case.

Comment (Jakub89) [reply concent_api/core/tests/test_auth_report_computed_task.py]: Okey, I'll switch it to other case.

Comment (rwrzesien) [reply concent_api/core/tests/test_auth_report_computed_task.py]: As discussed, it is ok to remove this as it is doing the same now as `# Step 9` of this test.

File: concent_api/core/message_handlers.py [hunk 5]
```diff
@@ -908,18 +908,21 @@ def store_subtask(
...
-def handle_messages_from_database(
-    client_public_key: bytes,
-    response_type: PendingResponse.Queue,
-) -> Union[message.Message, None]:
-    assert client_public_key    not in ['', None]
+def handle_messages_from_database(client_public_key: bytes) -> Union[message.Message, None]:
+    assert client_public_key not in ['', None]
 
     encoded_client_public_key = b64encode(client_public_key)
     pending_response = PendingResponse.objects.select_for_update().filter(
```
Comment (rwrzesien) [inline concent_api/core/message_handlers.py]: It would be the same and less redundant if you do:
```
pending_response = PendingResponse.objects.select_for_update().filter(
    client__public_key=encoded_client_public_key,
    delivered=False,
).order_by('created_at').first()
```

and remove the second query. This way it will return any `PendingResponse` it will find.
